bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -749,13 +770,27 @@ public synchronized ModuleHandlerFactory addingService(ServiceReference/* <Modul
             }
         }
         if (notInitailizedRules != null) {
-            for (String rUID : notInitailizedRules) {
-                setRule(rUID);
+            for (final String rUID : notInitailizedRules) {
+                scheduleRuleInitialization(rUID);
             }
         }
         return mhf;
     }
 
+    private synchronized void scheduleRuleInitialization(final String rUID) {
+        Future f = scheduleTasks.get(rUID);
+        if (f == null) {
+            ScheduledExecutorService ex = getScheduledExecutor();
+            f = ex.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    setRule(rUID);
+                }
+            }, 500, TimeUnit.MILLISECONDS);
+            scheduleTasks.put(rUID, f);
+        }
```

Comment:
How do you know that it is the last one? You only know that there is still activity. If your startup takes 30 seconds, you might now still run the job 60 times for the same rule.